### PR TITLE
feat: #1508 send email notification optionally

### PR DESCRIPTION
**Pull request by ianliuwk1019**
ref: #1508 
* Add param in schema for grant access and grant delegated admin endpionts to send email notification.
* Add util function to to check if the selected app is prod for prod deployment environment.
* Generate new api-client for endpoints' param addition.
* Add basic frontend checkbox component
* Add checkbox to grant access page and grant delegated admin page with default value.
* Frontend grant access/grant delegated admin to be able to call endpoint with the send email param.

### CODE REVIEW

File: frontend/src/services/utils.ts [hunk 1]
```diff
@@ -50,3 +52,8 @@ export const formValidationSchema = (isAbstractRoleSelected: boolean) => {
 export const isNewAccess = (newAccessIds: string[], accessId: number) => {
     return newAccessIds.includes(accessId.toString());
 };
+
+export const isSelectedAppProd_OnProdEnvironment = () => {
```
Comment (ianliuwk1019) [inline frontend/src/services/utils.ts:56]: Sorry for this odd looking variable name. I couldn't come up with a short and sweet name... normally I use camelCase, but I separate it with underscore to be a bit clear... if you come up with a better name please let me know...

File: frontend/src/components/common/BoolCheckbox.vue [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+interface CheckboxProps {
+    id?: string;
+    label?: string;
+    lfontSize?: string;
+    lfontWeight?: string;
+}
+const props = withDefaults(defineProps<CheckboxProps>(), {
+    id: "checkbox",
+    lfontSize: "0.875rem",
+    lfontWeight: "600"
```
Comment (MCatherine1994) [inline frontend/src/components/common/BoolCheckbox.vue:20]: Hi Ian, do we need these to be props? I feel we can add them when we need them, for now maybe then can just be local variables

File: frontend/src/services/EnvironmentSettings.ts [hunk 3]
```diff
@@ -66,6 +67,10 @@ export class EnvironmentSettings {
         return false
     }
 
+    isProdEnvironment() {
+        return window.localStorage.getItem(this.environmentDisplayNameKey) == ''
```
Comment (MCatherine1994) [inline frontend/src/services/EnvironmentSettings.ts:71]: I feel use display name to check environment feels weird... in the production environment, the "target_env" value will be "prod"

File: frontend/src/services/utils.ts (hunk 1 above)
Comment (MCatherine1994) [reply frontend/src/services/utils.ts:56]: isProdAppSelectedOnProdEnv?

Comment (ianliuwk1019) [reply frontend/src/services/utils.ts:56]: Sure, I can change to that

File: frontend/src/components/common/BoolCheckbox.vue (hunk 2 above)
Comment (ianliuwk1019) [reply frontend/src/components/common/BoolCheckbox.vue:20]: I see checkbox might soon be used in other places so adding them for convenience just for fontsize and fontweight. I can remove them, but I will keep the id (with default) if that's ok?

File: frontend/src/services/EnvironmentSettings.ts (hunk 3 above)
Comment (ianliuwk1019) [reply frontend/src/services/EnvironmentSettings.ts:71]: Yes, this seems to bit strange as I was thinking the same thing too. I saw previous one `isDevEnvironment` was using the same displayName and don't know why so just to keep it consistent. We can discuss this.

Comment (ianliuwk1019) [reply frontend/src/services/EnvironmentSettings.ts]: This is refactored.

File: frontend/src/components/common/BoolCheckbox.vue (hunk 2 above)
Comment (ianliuwk1019) [reply frontend/src/components/common/BoolCheckbox.vue]: Font ones are removed but kept the id.

**Comment by MCatherine1994** [comment]
Hi Ian, thanks for the update. Just want to double check the text color and font weight. It looks a little bit weird... 
Screen Shot 2024-08-08 at 4 34 20 PM

**Review by MCatherine1994 — APPROVED** [review]

**Comment by ianliuwk1019** [comment]
> Hi Ian, thanks for the update. Just want to double check the text color and font weight. It looks a little bit weird... Screen Shot 2024-08-08 at 4 34 20 PM



> Hi Ian, thanks for the update. Just want to double check the text color and font weight. It looks a little bit weird... Screen Shot 2024-08-08 at 4 34 20 PM

Yes, I make it a bit bold but the same color as other sidebar or page title color using `$light-text-secondary;` as it looks to me the mackup is a bit stand out than usual text.
image

**Pull request merged**